### ISSUE TITLE
don't run ci for files/directories irrelevant to CI

### ISSUE THREAD

**Issue by adriaanm** [scala/scabot#23]
- git config: gitconfig.SAMPLE .gitignore .mailmap 
 - dbuild: dbuild-meta.json 
 - validated by travis: .travis.yml Gemfile spec 
 - docs: CONTRIBUTING.md README.md docs/ tools/

(note that doc/ is relevant to integration build, it's included in scala-dist)

**Comment by retronym** [comment]
src/intellij src/eclipse

PRs with a "no-ci" label that we could add for other times when we know better.

**Comment by adriaanm** [comment]
Yep, no-ci could be good, but introduces a sneaky attack vector.

**Comment by retronym** [comment]
AFAICR, labels can only be added by team members.

**Comment by adriaanm** [comment]
true, but these days Scabot will likely beat you to building before you can add the label

**Comment by adriaanm** [comment]
unlike an issue, you can't add a label to a PR on creation

**Comment by retronym** [comment]
Perhaps it could cancel the builds upon creation of the label.

**Comment by retronym** [comment]
Not worth overcomplicating things for this use case, though.

**Comment by adriaanm** [comment]
Sure, but it seems we can catch 90% of the use cases by looking at the changed files, and we all know how hard it is to recall a launched missile :-)

**Comment by adriaanm** [comment]
For future reference, the commit api can tell you which files changed:

https://github.com/github/developer.github.com/blob/e33f8121defc22215425e8ee2f4ec2169b70bfba/lib/resources.rb#L526
